compute/cloud-client/src/test/java/compute/ipaddress/IPAddressTest.java

```diff
@@ -0,0 +1,308 @@
+    Assert.assertNotNull(addresses);
+    Assert.assertFalse(addresses.isEmpty());
+    Assert.assertTrue(addresses.stream().anyMatch(address
+            -> address.getName().equals(addressName)));
+
+    String regionAddressName = getNewAddressName(false);
+    addresses = ReserveNewExternalAddress
+            .reserveNewExternalIpAddress(PROJECT_ID, regionAddressName, false, true, REGION);
+    Assert.assertNotNull(addresses);
+    Assert.assertFalse(addresses.isEmpty());
+    Assert.assertTrue(addresses.stream().anyMatch(address
+            -> address.getName().equals(regionAddressName)));
+  }
+
+  @Test
+  public void assignStaticExternalNewVMAddressTest()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    String ipAddress = getExternalIpAddress(getNewAddressName(false), false);
+    String machineType = String.format("zones/%s/machineTypes/n1-standard-1", ZONE);
+    Instance instance = AssignStaticExternalNewVmAddress.assignStaticExternalNewVmAddress(
+            PROJECT_ID, EXTERNAL_NEW_VM_INSTANCE, ZONE, true, machineType, ipAddress);
+    Assert.assertNotNull(instance);
+    Assert.assertFalse(instance.getNetworkInterfacesList().isEmpty());
+    Assert.assertFalse(instance.getNetworkInterfacesList().get(0).getAccessConfigsList().isEmpty());
+    AccessConfig accessConfig = instance.getNetworkInterfacesList().get(0)
+            .getAccessConfigsList().get(0);
+    Assert.assertEquals(ipAddress, accessConfig.getNatIP());
+  }
+
+  @Test
+  public void assignStaticExistingVMAddressTest()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    Instance instance = AssignStaticExistingVm.assignStaticExistingVmAddress(
+            PROJECT_ID, EXTERNAL_NEW_VM_INSTANCE_2, ZONE, "nic0");
+    Assert.assertNotNull(instance);
+    Assert.assertFalse(instance.getNetworkInterfacesList().isEmpty());
+    Assert.assertFalse(instance.getNetworkInterfacesList().get(0).getAccessConfigsList().isEmpty());
+    AccessConfig accessConfig = instance.getNetworkInterfacesList().get(0)
+            .getAccessConfigsList().get(0);
+    Assert.assertTrue(accessConfig.getNatIP()
+            .matches("^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)\\.?\\b){4}$"));
+  }
+
+  @Test
+  public void promoteEphemeralIdTest()
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+
+    String ipAddress = null;
+    try (InstancesClient client = InstancesClient.create()) {
+      Instance instance = client.get(PROJECT_ID, ZONE, EXTERNAL_NEW_VM_INSTANCE_2);
+      for (NetworkInterface networkInterface : instance.getNetworkInterfacesList()) {
+        for (AccessConfig accessConfig : networkInterface.getAccessConfigsList()) {
+          if (accessConfig.getType().equals(AccessConfig.Type.ONE_TO_ONE_NAT.name())) {
+            ipAddress = accessConfig.getNatIP();
+            break;
```

Comment:
Can this be simplified with something like: `instance.getNetworkInterfacesList().getAccessConfigsList()`?

Reply:
No it can't be simplified, cause instance hase list of NetworkInterfaces and each NetworkInterface has list of AccessConfigs